tests/test_dashboard_server.py

```
@@ -1742,3 +1742,44 @@ def test_event_ticker_formats_timestamps_in_local_time():
     assert "Times shown in local time" in index_html
 
 
+@pytest.mark.skipif(NODE is None, reason="node not installed")
+def test_market_link_generates_safe_hyperlinks():
+    """marketLink generates safe anchor tags for markets with slug/url, escaping XSS."""
+    harness = Path(__file__).resolve().parent / "js" / "market_link_harness.js"
+    app_js = Path(__file__).resolve().parent.parent / "dashboard" / "static" / "app.js"
+
+    res = subprocess.run(
```

Ruff (0.16.2)

[error] `subprocess` call: check for execution of untrusted input

(S603)